### PR TITLE
TEN-173 removed getEntry function and just doing one fetch

### PR DESCRIPTION
**Pull request by Chrissystk**

### CODE REVIEW

**Comment by ChrisHillCap** [comment]
Ill give this a once over aswell in a mo

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    sessionData.get(REPORTED_FRAUD).map(_.as[Boolean])
+  val optUserAssignedSA =
+    sessionData.get(USER_ASSIGNED_SA_ENROLMENT).map(_.as[UsersAssignedEnrolment])
+  val optUserAssignedPT =
+    sessionData.get(USER_ASSIGNED_PT_ENROLMENT).map(_.as[UsersAssignedEnrolment])
+  val optOtherValidPTAAccounts =
+    sessionData.get(OTHER_VALID_PTA_ACCOUNTS).map(_.as[Seq[IVNinoStoreEntry]])
+  def optAccountDetails(credId: String): Option[AccountDetails] =
+    sessionData.get(accountDetailsForCredential(credId)).map(_.as[AccountDetails])
+
```
Comment (ChrisHillCap) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala:41]: can these have types

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import play.api.libs.json.JsValue
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.AccountTypes
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.models.{AccountDetails, IVNinoStoreEntry, UsersAssignedEnrolment}
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.models.forms.KeepAccessToSAThroughPTA
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.repository.SessionKeys.{ACCOUNT_TYPE, KEEP_ACCESS_TO_SA_THROUGH_PTA_FORM, OTHER_VALID_PTA_ACCOUNTS, REDIRECT_URL, REPORTED_FRAUD, USER_ASSIGNED_PT_ENROLMENT, USER_ASSIGNED_SA_ENROLMENT, accountDetailsForCredential}
+
+case class AccountDetailsFromMongo(accountType: AccountTypes.Value,
+                                   redirectUrl: String,
+                                   sessionData: Map[String, JsValue]) {
```
Comment (ChrisHillCap) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala:27]: should we make this val private, so it enforces people to create a new val and test it if they want to add a new key to the service

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    sessionData.get(accountDetailsForCredential(credId)).map(_.as[AccountDetails])
+
+}
+
+object AccountDetailsFromMongo {
+
+  val optAccountType = (sessionData: Map[String, JsValue]) =>
+    sessionData.get(ACCOUNT_TYPE).map(_.as[AccountTypes.Value])
+  val optRedirectUrl = (sessionData: Map[String, JsValue]) =>
+    sessionData.get(REDIRECT_URL).map(_.as[String])
```
Comment (ChrisHillCap) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala:49]: can these be typed

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/orchestrators/MultipleAccountsOrchestrator.scala [hunk 4]
```diff
@@ -221,36 +213,38 @@ class MultipleAccountsOrchestrator @Inject()(
...
-    def getSAAccountDetails: TEAFResult[Option[AccountDetails]] = {
-      if (!requestWithUserDetails.userDetails.hasSAEnrolment) {
-        eacdService.getUsersAssignedSAEnrolment(requestWithUserDetails, implicitly, implicitly).map(
-          user => user.enrolledCredential
-        ).flatMap {
-          case Some(userId) => usersGroupSearchService.getAccountDetails(userId)(implicitly, implicitly, requestWithUserDetails).map(Some(_))
-          case None => EitherT.right[TaxEnrolmentAssignmentErrors](Future.successful(Option.empty[AccountDetails]))
-        }
-      } else {
+    lazy val optSACredentialId: TEAFResult[Option[String]] =
```
Comment (ChrisHillCap) [inline app/uk/gov/hmrc/taxenrolmentassignmentfrontend/orchestrators/MultipleAccountsOrchestrator.scala:216]: i like that this is lazy

**Review by ChrisHillCap — CHANGES_REQUESTED** [review]
minor

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala (hunk 2 above)
Comment (Chrissystk) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala:27]: I'm not sure what you mean here?

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/orchestrators/MultipleAccountsOrchestrator.scala (hunk 4 above)
Comment (Chrissystk) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/orchestrators/MultipleAccountsOrchestrator.scala:216]: :-)

File: app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (Chrissystk) [reply app/uk/gov/hmrc/taxenrolmentassignmentfrontend/controllers/actions/AccountDetailsFromMongo.scala]: OK so not reading the sessionData directly but getting the data using the vals. I've pushed it up along with other changes.

**Review by ChrisHillCap — APPROVED** [review]

**Pull request merged**